adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -453,6 +526,25 @@ private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device a
         metrics.incrementMqttConnections(authenticatedDevice.getTenantId());
     }
 
+    private Future<MessageConsumer> createCommandConsumer(final Device authenticatedDevice) {
+        return getCommandConnection().getOrCreateCommandConsumer(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId(), (delivery, message) -> {
+            LOG.debug("command received for [{}] with delivery ", authenticatedDevice);
+            onCommandReceived(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId(), delivery, message);
+        }, close -> {
+            LOG.debug("command receiver link closed remotely"); // TODO reopen?
+            getCommandConnection().closeCommandReceiver(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId());
```

Comment:
IMHO better introduce a `CompositeFuture` for the close calls and collect the outcomes.

Reply:
But why are all reponse senders implicitly closed when the command receiver link was closed (remotely)?
Is that resonable?

Reply:
I agree that it would be better to stay with the response senders until the device disconnects. Or close after each send message - this depends on link creation vs. link count costs.